### ISSUE TITLE
Use Primordials for accessing ".buffer" properties in objects

### ISSUE THREAD

**Issue by CrimsonCodes0** [denoland/deno#11355]
> I mentioned that the WebGPU PR left a few dynamic object accesses. Ref https://github.com/denoland/deno/pull/11265#issuecomment-873629656
>
> While looking through it, the changes are hard to do as they primarily accept WebIDL `BufferSource` objects, which themselves are very dynamic objects. Comparing the prototype against via the `instanceof` operator is a very fallible operation in that it doesn't genuinely give the true type of the object, therefore checking the prototype and then using an accessor such as `TypedArrayPrototypeGetBuffer` or `DataViewPrototypeGetBuffer` can still throw. In those cases, there's not much more that can be done, besides possibly trying the opposite accessor, but then it may have just thrown due to other reasons anyways. Do you have any suggestions as to how this should be handled?
>
> [webidl.js](https://github.com/denoland/deno/blob/main/extensions/webidl/00_webidl.js) is used across the entire codebase, yet it also suffers from this vulnerability as it doesn't use the prototype primordials for these.

_Originally posted by @CrimsonCodes0 in https://github.com/denoland/deno/issues/11224#issuecomment-874469585_

I could easily implement a small utility/helper function for accessing the buffer of an object that has been verified to be a WebIDL BufferSource via primordials.

Somewhat related, because I'd have naturally used something like this in the function's implementation, some of the internal code might be best written with assertions that abort the entire process. E.g. HTML spec assertions are use in a way that they are intended to be completely unreachable, and can be entirely eluded by implementers.

For example, the Request class's implementation has [an assertion](https://github.com/denoland/deno/blob/5fa58c92165e23386b8ed3c3079103997fe1bef9/extensions/fetch/23_request.js#L238) that simply throws a catch-able JavaScript error. Might I suggest that `Deno.exit` or something of the sort is called to indicate that the internal code should have never been reached (analogous to a Rust `unreachable!(...)`)?

Otherwise usage of the primordials for these situations would be as simple as checking that object is of the correct type (which is already done), and then calling the helper on the object to receive it's internal `ArrayBuffer`. (I'd be happy to rewrite any code to use primordials for buffer access too :))

**Comment by CrimsonCodes0** [comment]
Ah, I forgot about this, I'll get around it soon. (Currently blocked on SharedArrayBuffer prototype being in the snapshot)
